csrc/ops/alias.cpp

```diff
@@ -205,26 +205,45 @@ TensorView* flatten(TensorView* x, int64_t start_dim, int64_t end_dim) {
   return out;
 }
 
-TensorView* squeeze(TensorView* x, const std::vector<bool>& to_squeeze) {
+TensorView* squeeze(TensorView* x, const std::vector<int64_t>& dims) {
   NVF_ERROR(x != nullptr, "Input is invalid.");
   auto x_dom = x->domain()->noReductions();
   const auto ndims = static_cast<int>(x_dom.size());
 
   NVF_ERROR(
-      ndims == (int)to_squeeze.size(),
-      "Invalid to_squeeze for squeeze: ",
-      to_squeeze,
-      ". Input tensor: ",
-      x->toString());
+      (int)dims.size() <= ndims,
+      "The dims to squeeze must be <= the number of dims of the input tensorx. ",
+      "Squeeze dims: ",
+      dims.size(),
+      " Input Tensor dims: ",
+      ndims);
+
+  std::vector<bool> to_squeeze(ndims, false);
+  for (auto dim : dims) {
+    // Handle negative relative to the end dimensions specifications
+    if (dim < 0) {
+      dim = static_cast<int64_t>(to_squeeze.size()) + dim;
+    }
+    NVF_CHECK(
+        (dim >= 0) && (static_cast<size_t>(dim) < to_squeeze.size()),
+        "Squeeze dim is outside of Tensor size! Tensor Size: ",
+        to_squeeze.size(),
+        " Dim: ",
+        dim);
+    to_squeeze[dim] = true;
+  }
 
   std::vector<IterDomain*> out_domain;
   for (const auto idx : c10::irange(ndims)) {
     auto id = x_dom[idx];
     if (to_squeeze[idx]) {
       if (!id->isSymbolic()) {
-        NVF_CHECK(
-            id->isBroadcast(),
-            "Can not squeeze non-broadcasting dimension(s).");
+        // If a squeeze is attempted on a non-broadcast dimension
+        // just don't do it!  This confirms with Pytorch.
+        if (!id->isBroadcast()) {
```

Comment:
I'm wondering what happens when we try to squeeze a reshaped size-1 dimension. I think we'll accidentally miss that squeeze, since here we only squeeze a dimension if it's not broadcast.

Can we verify if it indeed gives us a rank-2 tensor in the example below?

```
        inputs = [
            torch.randn((3,), dtype=torch.float16, device="cuda:0").as_strided(
                (3, 4), (1, 0)
            ),
        ]
        def fusion_func(fd: FusionDefinition) -> None:
            T2 = fd.define_tensor(
                shape=[-1, -1],
                contiguity=[None, True],
                dtype=DataType.Half,
                is_cpu=False,
            )
            S7 = fd.define_scalar(3, dtype=DataType.Int)
            S8 = fd.define_scalar(4, dtype=DataType.Int)
            S9 = fd.define_scalar(1, dtype=DataType.Int)
            V10 = fd.define_vector([S7, S8, S9], dtype=DataType.Int)
            T11 = fd.ops.reshape(T2, new_shape=V10)
            T12 = fd.ops.squeeze(T11, (2))

        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
```

Reply:
tagging issue #1393 

I'm expecting the squeeze would fail the same thing. I'm wondering if we are doing the right thing with our dynamic reshape, if we need to patch everything c++ API to consider the symbolic dimension....

Reply:
There seems to be more than one issue in this case related to just expanding. 
```
import torch
from nvfuser import FusionDefinition, DataType

inputs = [
        torch.randn((3,), dtype=torch.float16, device="cuda:0").as_strided( (3, 4), (1, 0) ),
         ]

def fusion_func(fd: FusionDefinition) -> None:
    T2 = fd.define_tensor( shape=[-1, -1], contiguity=[None, True], dtype=DataType.Half, is_cpu=False, )
    S7 = fd.define_scalar(3, dtype=DataType.Int)
    S8 = fd.define_scalar(4, dtype=DataType.Int)
    S9 = fd.define_scalar(1, dtype=DataType.Int)
    V10 = fd.define_vector([S7, S8, S9], dtype=DataType.Int)
    T11 = fd.ops.reshape(T2, new_shape=V10)
    #T12 = fd.ops.squeeze(T11, (2,))
    fd.add_output(T11)

with FusionDefinition() as fd:
    fusion_func(fd)

out = fd.execute(inputs)
```

Error:
```
Traceback (most recent call last):
  File "/opt/pytorch/nvfuser/nvfuser/__init__.py", line 137, in execute
    result = self._execute(
RuntimeError: t.size(i) == 1 || t.stride(i) == 0 INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/expr_evaluator.cpp":130, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. IterDomain bS0{i0} in TensorView T0_g[ bS0{i0}, iS1{i1} ] has expanded extent but input tensor has size 3 and stride 1 in dimension 0
```

Reply:
sorry this is a busted example that I put there.
Contiguity should be `[True, None]` instead. And then there seems to be an issue with dynamic reshape not considering the expanded extent when validating consistent numel.

I think Jingyue(?!) has already fixed that and I see the updated code in main.